pinot-common/src/main/java/org/apache/pinot/common/request/context/ExpressionContext.java

```diff
@@ -94,7 +113,7 @@ public boolean equals(Object o) {
       return false;
     }
     ExpressionContext that = (ExpressionContext) o;
-    return _type == that._type && Objects.equals(_value, that._value) && Objects.equals(_function, that._function);
+    return _type.equals(that._type) && Objects.equals(_identifier, that._identifier) && Objects.equals(_function, that._function) && Objects.equals(_literal, that._literal);
```

Comment:
this might have backward compatibility issue? e.g. if `_value` was still storing literal in broker but server upgraded to newer version and stored literal in `_literal` field.